### PR TITLE
FB-748

### PR DESCRIPTION
**Pull request by sequencemedialimited**
[FB-748](https://dsdmoj.atlassian.net/browse/FB-748) - **Checkbox data missing 'key' in Runner when submitted**

**In scope**:
The PDF generator was consuming the `answers` collection which is prepared for "check answers" pages, but there's an incompatibility with the way that Checkboxes components are represented in that collection and the needs of the PDF generator. (None of the checkbox input names was being captured in the `answers` list, and the values of several checkbox inputs were concatenated together.)

I've created a _second_ collection in which individual checkbox inputs are represented with their name and value together, `answersList`.

**Not in scope**:
- There is some real nastiness in the `getDisplayValue` and `getRedactedValue` mechanisms which has an impact on the code which calls them. I have let some of that be for now
- The consequence is some duplication for absurdly minor reasons
- I have left some tricky bits in place because there's no obvious remedy (cost v. benefit)

### CODE REVIEW

File: lib/controller/component/type/answers/answers.answers.js [hunk 1]
```diff
@@ -46,10 +46,7 @@ function reduceHeadings (accumulator, answer, index, array) {
   return accumulator.concat(answer)
 }
 
-function getAnswers (componentInstance, userData, pageInstance) {
-  const parsedComponentInstance = JSON.parse(JSON.stringify(componentInstance))
```
Comment (cob16) [inline lib/controller/component/type/answers/answers.answers.js]: interesting. I wonder how manny times we do this in the codebace

File: lib/presenter/pdf-payload.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ const sections = (section) => {
 const questions = (section) => {
   return (section.answers || []).map(answer => {
     return {
-      label: answer.key.html,
+      label: answer.key.text,
```
[the remark this answers is not in the document]
Comment (sequencemedialimited) [reply lib/presenter/pdf-payload.js]: They do now

File: lib/presenter/pdf-payload.js [hunk 3]
```diff
@@ -8,7 +8,7 @@ const pdfPayload = (submission) => {
 }
 
 const sections = (section) => {
-  return section.answers.map(sectionAnswer => {
+  return (section.answersList || []).map(sectionAnswer => {
```
[the remark this answers is not in the document]
Comment (sequencemedialimited) [reply lib/presenter/pdf-payload.js]: Hm, will it? If `section.answers` isn't defined JS will throw a `ReferenceError` on that line

I'm assuming it has never _not_ encountered an array

File: lib/page/update-control-names/update-control-names.js [hunk 4]
```diff
@@ -6,17 +6,28 @@ const getComponentComposite = require('../../controller/component/get-composite'
 
 const getRedactedValue = (nameInstance, userData, skipRedaction, scope) => {
   let value = userData.getUserDataProperty(nameInstance.name, undefined, scope)
-  if (!value || skipRedaction) {
+
+  /*
+   *  Sigh
```
[the remark this answers is not in the document]
Comment (sequencemedialimited) [reply lib/page/update-control-names/update-control-names.js]: Yeah, I'll kill it

File: lib/controller/component/type/answers/answers.answers.js (hunk 1 above)
Comment (sequencemedialimited) [reply lib/controller/component/type/answers/answers.answers.js]: LOTS

File: lib/page/update-control-names/update-control-names.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (sequencemedialimited) [reply lib/page/update-control-names/update-control-names.js]: I need to figure out a way of logging these observations that doesn't involve littering the code with comments or creating tons of teeny tiny stories to fix them

File: lib/presenter/pdf-payload.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sequencemedialimited) [reply lib/presenter/pdf-payload.js]: I'll kill it ;)

File: lib/controller/component/type/answers/answers.answers.js (hunk 1 above)
Comment (sequencemedialimited) [reply lib/controller/component/type/answers/answers.answers.js]: It's completely unnecessary here. The code beyond this only picks off two values from `componentInstance`, both booleans

**Comment by sequencemedialimited** [comment]
Yep, I'll make sure to put the title in next time

**Pull request merged**